archunit-integration-test/src/test/java/com/tngtech/archunit/ImporterRules.java

```diff
@@ -3,22 +3,39 @@
 import com.tngtech.archunit.base.DescribedPredicate;
 import com.tngtech.archunit.core.domain.JavaClass;
 import com.tngtech.archunit.core.domain.JavaClassDescriptor;
+import com.tngtech.archunit.core.domain.JavaModifier;
 import com.tngtech.archunit.core.importer.ClassFileImporter;
 import com.tngtech.archunit.core.importer.DomainBuilders;
 import com.tngtech.archunit.junit.ArchTest;
 import com.tngtech.archunit.lang.ArchRule;
 
 import static com.tngtech.archunit.ArchUnitArchitectureTest.THIRDPARTY_PACKAGE_IDENTIFIER;
+import static com.tngtech.archunit.base.DescribedPredicate.doNot;
 import static com.tngtech.archunit.base.DescribedPredicate.not;
 import static com.tngtech.archunit.core.domain.JavaClass.Predicates.belongToAnyOf;
+import static com.tngtech.archunit.core.domain.JavaClass.Predicates.resideOutsideOfPackage;
 import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.noClasses;
 
 public class ImporterRules {
 
     @ArchTest
     public static final ArchRule domain_does_not_access_importer =
             noClasses().that().resideInAPackage("..core.domain..")
-                    .should().accessClassesThat(belong_to_the_import_context());
+                    .should().dependOnClassesThat(belong_to_the_import_context());
+
+    @ArchTest
+    public static final ArchRule asm_is_only_used_in_importer_or_JavaClassDescriptor =
+            noClasses()
+                    .that(
+                            resideOutsideOfPackage("..core.importer..")
+                                    .and(doNot(belongToAnyOf(
+                                            JavaClassDescriptor.class,
+                                            // whenever we get the chance to break the public API
+                                            // we should likely remove the dependencies from JavaModifier to ASM.
+                                            // Those crept in by accident at some point because the design was convenient.
+                                            JavaModifier.class)))
```

Comment:
We might not even have to exclude `JavaModifier` as long as we only import static `int` constants, cf. https://github.com/TNG/ArchUnit/issues/298#issuecomment-577038056.

Reply:
Okay, as agreed I'm only gonna add the comment there, but not add `JavaModifier` to avoid further dependencies slipping in :slightly_smiling_face: